### PR TITLE
Add factory tests

### PR DESCRIPTION
**Pull request by prolic**
see: https://github.com/prooph/event-store/issues/70

### CODE REVIEW

File: src/Container/EventStoreFactory.php [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <?php
 /*
- * This file is part of the prooph/proophessor.
- * (c) 2014 - 2015 prooph software GmbH <contact@prooph.de>
+ * This file is part of the prooph/event-store.
+ * (c) Alexander Miertsch <contact@prooph.de>
```
Comment (codeliner) [inline src/Container/EventStoreFactory.php]: oh, good catch. The header of the event-store is an old one. The copyright of proophessor is the newer version. But I create an ticket to cover the issue.

File: tests/Container/EventStoreFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        $adapterMock = $this->getMockForAbstractClass(Adapter::class);
+
+        $containerMock = $this->getMockForAbstractClass(ContainerInterface::class);
+        $containerMock->expects($this->at(0))->method('get')->with('config')->willReturn($config);
+        $containerMock->expects($this->at(1))->method('get')->with('adapter')->willReturn($adapterMock);
+
+        $factory = new EventStoreFactory();
+        $eventStore = $factory($containerMock);
+
+        $this->assertInstanceOf(EventStore::class, $eventStore);
```
Comment (codeliner) [inline tests/Container/EventStoreFactoryTest.php]: We should also test that adapter is really injected:
```php
$this->assertSame($adapterMock, $eventStore->getAdapter());
```

File: tests/Container/EventStoreFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        $eventEmitterMock = $this->getMockForAbstractClass(ActionEventEmitter::class);
+
+        $containerMock = $this->getMockForAbstractClass(ContainerInterface::class);
+        $containerMock->expects($this->at(0))->method('get')->with('config')->willReturn($config);
+        $containerMock->expects($this->at(1))->method('get')->with('event_emitter')->willReturn($eventEmitterMock);
+
+        $factory = new EventStoreFactory();
+        $eventStore = $factory($containerMock);
+
+        $this->assertInstanceOf(EventStore::class, $eventStore);
```
Comment (codeliner) [inline tests/Container/EventStoreFactoryTest.php]: same as above

```php
$this->assertSame($eventEmitterMock, $eventStore->getActionEventEmitter());
```

File: tests/Container/EventStoreFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        $featureMock = $this->getMockForAbstractClass(Feature::class);
+
+        $containerMock = $this->getMockForAbstractClass(ContainerInterface::class);
+        $containerMock->expects($this->at(0))->method('get')->with('config')->willReturn($config);
+        $containerMock->expects($this->at(1))->method('get')->with('plugin')->willReturn($featureMock);
+
+        $factory = new EventStoreFactory();
+        $eventStore = $factory($containerMock);
+
+        $this->assertInstanceOf(EventStore::class, $eventStore);
```
Comment (codeliner) [inline tests/Container/EventStoreFactoryTest.php]: same as above

here we can expect that the `setUp` method of the `$featureMock` is inoked

**Comment by prolic** [comment]
done.

**Comment by codeliner** [comment]
:+1:

**Pull request merged**